### PR TITLE
spring-boot: init at 2.1.9

### PR DESCRIPTION
**Pull request by moaxcp**
###### Motivation for this change

Adding spring-boot cli

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executable appears to work

```
[3 built, 28 copied (265.4 MiB), 93.5 MiB DL]
https://github.com/NixOS/nixpkgs/pull/70282
1 package were build:
spring-boot
```
```
[nix-shell:~/.cache/nix-review/pr-70282]$ nix path-info -Sh ./results/spring-boot
/nix/store/2znr28c2p56z9hpbdj7cckzb7wa0zwav-spring-boot-2.1.8    483.5M
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build spring-boot

File: pkgs/development/tools/spring-boot/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  src = fetchzip {
+    url = "https://repo.spring.io/release/org/springframework/boot/${pname}-cli/${version}.RELEASE/${pname}-cli-${version}.RELEASE-bin.zip";
+    sha256 = "0shkznvv8abb365pgzfp265ihmf17mqigzzimdkzyp9jrh9kgiqx";
+  };
+
+  nativeBuildInputs = [ makeWrapper installShellFiles ];
+
+  installPhase = ''
+    rm bin/spring.bat
```
Comment (c0bw3b) [inline pkgs/development/tools/spring-boot/default.nix]:
```suggestion
    runHook preInstall
    rm bin/spring.bat
```

File: pkgs/development/tools/spring-boot/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    installShellCompletion --bash shell-completion/bash/spring
+    installShellCompletion --zsh shell-completion/zsh/_spring
+    rm -r shell-completion
+    cp -r . $out
+    wrapProgram $out/bin/spring \
+      --prefix JAVA_HOME : ${jdk}
+  '';
```
Comment (c0bw3b) [inline pkgs/development/tools/spring-boot/default.nix]:
```suggestion
  runHook postInstall
  '';
```

File: pkgs/development/tools/spring-boot/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  meta = with stdenv.lib; {
+    description = ''
```
Comment (c0bw3b) [inline pkgs/development/tools/spring-boot/default.nix]: The description might be shorter. From [the manual](https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes):

> Don’t include a period at the end. Don’t include newline characters. Capitalise the first character. For brevity, don’t repeat the name of package — just describe what it does.

**Review by c0bw3b — COMMENTED** [review]
v2.1.9 has been published in the meantime.

File: pkgs/development/tools/spring-boot/default.nix (hunk 1 above)
Comment (moaxcp) [reply pkgs/development/tools/spring-boot/default.nix]: What features do preInstall and postInstall provide that is needed in this build? I have a few other similar packages that do not run these either.

File: pkgs/development/tools/spring-boot/default.nix (hunk 3 above)
Comment (moaxcp) [reply pkgs/development/tools/spring-boot/default.nix]: Thanks for pointing out the manual.

File: pkgs/development/tools/spring-boot/default.nix (hunk 1 above)
Comment (moaxcp) [reply pkgs/development/tools/spring-boot/default.nix]: I added these changes.

**Comment by moaxcp** [comment]
I updated to 2.1.9

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build spring-boot

**Review by c0bw3b — APPROVED** [review]
Result of `nix-review pr 70282` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>1 package were build:</summary>

  - spring-boot
</details>

```shell
$ ./results/spring-boot/bin/spring shell
Spring Boot (v2.1.9.RELEASE)
Hit TAB to complete. Type 'help' and hit RETURN for help, and 'exit' to quit.

$ version
Spring CLI v2.1.9.RELEASE
$ encodepassword mysecret
{bcrypt}$2a$10$/SM56QEHnD3FTwoI18zv2eH.ZPrIHlrXtySof7iijwzQ.9vqO6acu
```

**Pull request merged**